### PR TITLE
Fix #679 updated realpath util method

### LINKED ISSUE #679: Error Occurred while creating pdf from excel.
Error Occurred while creating pdf from excel.

### Steps to reproduce
Click on Excel to Pdf in the drawer 
Select Excel file from google drive path.
Click excel to pdf and enter pdf name.

### Expected behaviour
Pdf should be generated.

### Actual behaviour
Error message  because file path is null in Excel to pdf fragment if file is selected from drive.

### PR DESCRIPTION
**Pull request by jddeep**
# Description
updated realpath util method.

Fixes #679 

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by jddeep** [comment]
@Swati4star Please review.

**Comment by jddeep** [comment]
@Swati4star If you still are facing any error to load the files, please log the actual uri's of the files.And please let me know.

**Comment by Swati4star** [comment]
@vyankatesh24 Can you test it locally once?

**Comment by jddeep** [comment]
@vyankatesh24 Can you log the full URI of the file. I mean just log the `mUri` and let me know about it.

**Comment by vyankatesh24** [comment]
`content://com.mi.android.globalFileexplorer.myprovider/external_files/1.xlsx`

**Comment by jddeep** [comment]
@vyankatesh24 Please review now.

**Comment by jddeep** [comment]
And @vyankatesh24 Just to be sure your `external_files` from the uri path is a folder?

**Comment by vyankatesh24** [comment]
> And @vyankatesh24 Just to be sure your `external_files` from the uri path is a folder?

No!

**Comment by jddeep** [comment]
> `content://com.mi.android.globalFileexplorer.myprovider/external_files/1.xlsx`

@vyankatesh24 Then what does this mean? Does this mean there is `1.xlsx` in your base root directory?

**Comment by vyankatesh24** [comment]
> > `content://com.mi.android.globalFileexplorer.myprovider/external_files/1.xlsx`
> 
> @vyankatesh24 Then what does this mean? Does this mean there is `1.xlsx` in your base root directory?

yes!

**Comment by jddeep** [comment]
@vyankatesh24 Please check again.

**Comment by jddeep** [comment]
@vyankatesh24 Also can you please log your `mRealPath` and tell me what do you actually get there?

**Comment by vyankatesh24** [comment]
> @vyankatesh24 Also can you please log your `mRealPath` and tell me what do you actually get there?

`/storage/emulated/0/external_files/1.xlsx`

**Comment by jddeep** [comment]
@vyankatesh24 Did my latest change work though?

**Comment by jddeep** [comment]
@vyankatesh24 Please confirm if my last commit/push work? If not I will be pushing a new change.

**Comment by vyankatesh24** [comment]
> @vyankatesh24 Please confirm if my last commit/push work? If not I will be pushing a new change.

No

**Comment by jddeep** [comment]
@vyankatesh24 Made some changes. Please check.

**Comment by jddeep** [comment]
@vyankatesh24 If this doesn't work too, please let me know about the exact uri/path you used to test the `PPT` feature.

**Comment by vyankatesh24** [comment]
@jddeep 
Conversion working for  `content://com.mi.android.globalFileexplorer.myprovider/external_files/1.xlsx` :+1: 
but not for `content://com.google.android.apps.docs.storage.legacy/enc%3Dixdcs7oqHCZwyMukpyuhnbcvnSWsXQMyrmxbfmkc2Z5R_kHLkQ1haWfePHtDkaueDo4i-mhZKKzi%0AXO9hOzjFB0LrtzrrQxdgequn2XJIEJK-kPVd3krPDj9xNDIESFImnoNwSQ%3D%3D%0A`

**Comment by vyankatesh24** [comment]
Also not working for this
URI: `content://com.mi.android.globalFileexplorer.myprovider/external_files/Download/CounterFoil_260418.xls`

Output: `/storage/emulated/0/Download`

**Comment by jddeep** [comment]
> Also not working for this
> URI: `content://com.mi.android.globalFileexplorer.myprovider/external_files/Download/CounterFoil_260418.xls`
> 
> Output: `/storage/emulated/0/Download`

@vyankatesh24 Now this time this should work for all.

**Comment by jddeep** [comment]
@vyankatesh24 Please review.

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java [hunk 1]
```diff
@@ -113,6 +120,13 @@ public static boolean isExternalStorageDocument(Uri uri) {
         return "com.android.externalstorage.documents".equals(uri.getAuthority());
     }
 
+    public static StringBuilder trimExternal(String path) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: move this function to StringUtils class & add comments to it

**Comment by jddeep** [comment]
@Swati4star Done. Please review.

**Comment by jddeep** [comment]
@vyankatesh24 I think we cannot upload drive related files now. We need some more support for that. But Now Any kind of file uploads from the device locally can be feasible as you rightly mentioned, 👍

**Comment by vyankatesh24** [comment]
> @vyankatesh24 I think we cannot upload drive files as of now. We need some more support for that. But Now Any kind of file uploads from the device locally can be feasible as you rightly mentioned,

Then handle the exception for such URIs. The app should not crash.

**Comment by jddeep** [comment]
> app

Alright, that can be done. 👍

**Comment by jddeep** [comment]
@vyankatesh24 Done! Please review.

**Comment by vyankatesh24** [comment]
Well done :+1: @jddeep

**Comment by worm69** [comment]
Hi app can handle this type uri? from sdcard with following android emulator
image


image

**Comment by jddeep** [comment]
Yes @worm69 the App can handle most types of URI's. But some specific drive file uri's are not handled at this moment.

**Pull request merged**